File: src/Client/Logistics.Client/Models/CreateLoad.cs
using System.ComponentModel.DataAnnotations;
using Logistics.Client.Constants;

namespace Logistics.Client.Models;

public record CreateLoad
{
    public string? Name { get; set; }

    [Required]
    public string? SourceAddress { get; set; }

    [Required]
    public string? DestinationAddress { get; set; }

    [Required]
    [Range(LoadConsts.MinDeliveryCost, LoadConsts.MaxDeliveryCost)]
    public double DeliveryCost { get; set; }

    [Required]
    public double Distance { get; set; }

    [Required]
    public string? AssignedDispatcherId { get; set; }

    [Required]
    public string? AssignedDriverId { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/load/CreateLoad_.ts
import {Address} from '../Address_';

export interface CreateLoad {
  name?: string;
  originAddress: Address;
  originAddressLat: number;
  originAddressLong: number;
  destinationAddress: Address;
  destinationAddressLat: number;
  destinationAddressLong: number;
  deliveryCost: number;
  distance: number;
  assignedDispatcherId: string;
  assignedTruckId: string;
  customerId: string;
}
